### PR TITLE
Fix: parentNode error.

### PR DESCRIPTION
**Pull request by tiaguinho**
This validations fix the problem with "Enter" inside an iframe.
I have this issue using ckeditor inside an Angular project.

I think this PR also fix [#930](https://github.com/ckeditor/ckeditor5/issues/930)

## Adicional information

Because the main project isn't an Angular project,  my project (let's call FOO) have to live inside an iframe.
To make sure that our preview isn't affected by FOO style, the preview also live inside an iframe. So the structure is like this:
```typescript
MAIN PROJECT:
   - FOO iframe:
      - PREVIEW iframe (ckeditor live's here)
```

When we press enter, backspace or delete, the error ```Uncaught TypeError: Cannot read property 'parentNode' of null``` is throw.

I have tested this PR proposal inside FOO and the error was gone.

### CODE REVIEW

**Comment by tiaguinho** [comment]
@Reinmar can you take a look?

**Comment by Reinmar** [comment]
Hi! Thanks for the PR. I'm leaving it for @oleq to review, because he knows this code better. 

However, to merge this change we need to have an automated test verifying the patch and the code coverage back to 100%. Ideally, 

You can read about our testing env in https://docs.ckeditor.com/ckeditor5/latest/framework/guides/contributing/testing-environment.html.

File: src/dom/scroll.js [hunk 1]
```diff
@@ -45,29 +45,33 @@ export function scrollViewportToShowTarget( { target, viewportOffset = 0 } ) {
...
-		scrollWindowToShowRect( currentWindow, targetRect, viewportOffset );
-
-		if ( currentWindow.parent != currentWindow ) {
-			// Keep the reference to the <iframe> element the "previous current window" was
-			// rendered within. It will be useful to re–calculate the rect of the target
-			// in the parent window's relative geometry. The target's rect must be shifted
-			// by it's iframe's position.
-			currentFrame = currentWindow.frameElement;
-			currentWindow = currentWindow.parent;
+		if ( firstAncestorToScroll ) {
```
Comment (oleq) [inline src/dom/scroll.js]: What does this condition actually do? In which cases there's no `firstAncestorToScroll`? 

As I [wrote](https://github.com/ckeditor/ckeditor5/issues/930#issuecomment-377495214) I tested the editor inside an iframe (not JSFiddle), and I couldn't reproduce the issue. What makes your use–case different?

This condition disables scrolling (ancestors, window) completely, which is wrong IMO. I see no good reason why it should be disabled; the bugs should be fixed instead.

Comment (tiaguinho) [reply src/dom/scroll.js]: When I start debugging ckeditor inside my application, I get the following result when I press ```enter```:

* ```currentFrame``` equal null;
* ```currentWindow``` different of ```targetWindow```;
* ```getParentElement``` is called with a null param;
* console output error when the ```getParentElement``` try to return ***parentNode*** of null.

The changes I have made just make sure that ```elementOrRange``` isn't null.

**Comment by tiaguinho** [comment]
@oleq I don't know if the problem is related to the fact that my Angular app is running inside an iframe, but I rebuild ckeditor using my fork and the error was fixed.

**Comment by Reinmar** [comment]
@oleq, I've just encountered https://github.com/ckeditor/ckeditor5/issues/930 and we should investigate this.

**Comment by oleq** [comment]
@tiaguinho I found the root of the issue and moved the code to a different PR because I was unable to contribute to this one. Can you confirm that it works for you in https://github.com/ckeditor/ckeditor5-utils/pull/242?

**Comment by Reinmar** [comment]
I merged https://github.com/ckeditor/ckeditor5-utils/pull/242 already, so if you'll find that the issue still occurs in your case, please report a new ticket.

**Pull request closed without merging**